### ISSUE TITLE
Pytorch 1.4 compilation hangs on AMD Epyc

### ISSUE THREAD

**Issue by surak** [pytorch/pytorch#40968]
## 🐛 Bug

[1352/3291] : && /software/CMake/3.14.0-GCCcore-8.3.0/bin/cmake -E remove lib/libfbgemm.a && /software/binutils/2.32-GCCcore-8.3.0/bin/ar qc lib/libfbgemm.a  third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/ExecuteKernel.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/ExecuteKernelU8S8.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/Fbgemm.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/FbgemmFP16.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/FbgemmConv.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/FbgemmI64.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/FbgemmI8Spmdm.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/FbgemmSpConv.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/FbgemmSpMM.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/Fused8BitRowwiseEmbeddingLookup.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/GenerateKernelU8S8S32ACC16.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/GenerateKernelU8S8S32ACC16Avx512.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/GenerateKernelU8S8S32ACC16Avx512VNNI.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/GenerateKernelU8S8S32ACC32.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/GenerateKernelU8S8S32ACC32Avx512.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/GenerateKernelU8S8S32ACC32Avx512VNNI.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/GroupwiseConvAcc32Avx2.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/PackAMatrix.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/PackAWithIm2Col.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/PackBMatrix.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/PackMatrix.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/PackAWithQuantRowOffset.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/PackAWithRowOffset.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/PackWeightMatrixForGConv.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/PackWeightsForConv.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/QuantUtils.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/RefImplementations.cc.o third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/Utils.cc.o third_party/fbgemm/CMakeFiles/fbgemm_avx2.dir/src/FbgemmFP16UKernelsAvx2.cc.o third_party/fbgemm/CMakeFiles/fbgemm_avx2.dir/src/FbgemmI8Depthwise3DAvx2.cc.o third_party/fbgemm/CMakeFiles/fbgemm_avx2.dir/src/FbgemmI8Depthwise3x3Avx2.cc.o third_party/fbgemm/CMakeFiles/fbgemm_avx2.dir/src/FbgemmI8DepthwiseAvx2.cc.o third_party/fbgemm/CMakeFiles/fbgemm_avx2.dir/src/FbgemmI8DepthwisePerChannelQuantAvx2.cc.o third_party/fbgemm/CMakeFiles/fbgemm_avx2.dir/src/OptimizedKernelsAvx2.cc.o third_party/fbgemm/CMakeFiles/fbgemm_avx2.dir/src/PackDepthwiseConvMatrixAvx2.cc.o third_party/fbgemm/CMakeFiles/fbgemm_avx2.dir/src/QuantUtilsAvx2.cc.o third_party/fbgemm/CMakeFiles/fbgemm_avx2.dir/src/U 

## To Reproduce

1. Compile pytorch from source on AMD Epyc.

it hangs mid-compilation, as seen abot.

## Expected behavior

An error message, at least.

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0): 1.4
 - OS (e.g., Linux): Centos 7
 - How you installed PyTorch (`conda`, `pip`, source): source
 - Build command you used (if compiling from source): easybuild recipe
 - Python version: 3.6.8
 - CUDA/cuDNN version: 7.5.1.10
 - GPU models and configuration: nvidia v100
 - Any other relevant information: jusuf supercomputer from Jülich Supercomputing Centre, Germany

## Additional context

It's the exact same recipe used to compile Pytorch 1.4 on every other platform on the institute. The only difference is the cpu.

**Comment by emilwallner** [comment]
Any update on this? Does it work on previous Pytorch versions?

**Comment by surak** [comment]
no, 1.1, 1.4 and 1.5 fail.

**Comment by surak** [comment]
And 1.6 fails too.

**Comment by dskhudia** [comment]
This seems to be the gcc bug that @malfet reported and gcc was patched. Please try with updated gcc.

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=97028

**Comment by surak** [comment]
In any case, @malfet submitted a patch to FBGEMM which should have prevented this from showing up, no? https://github.com/pytorch/FBGEMM/pull/428

**Comment by malfet** [comment]
Yes, it should be fixed in 1.7, can someone please confirm that?

**Comment by dskhudia** [comment]
> In any case, @malfet submitted a patch to FBGEMM which should have prevented this from showing up, no? [pytorch/FBGEMM#428](https://github.com/pytorch/FBGEMM/pull/428)

Is this patch included in the fbgemm code you are using?
